vars/publishSmokeTestResults.groovy

```diff
@@ -0,0 +1,256 @@
+                            "distribution_build_url": {
+                                "type": "keyword"
+                            },
+                            "platform": {
+                                "type": "keyword"
+                            },
+                            "qualifier": {
+                                "type": "keyword"
+                            },
+                            "rc": {
+                                "type": "keyword"
+                            },
+                            "rc_number": {
+                                "type": "integer"
+                            },
+                            "smoke_test_build_number": {
+                                "type": "integer"
+                            },
+                            "smoke_test_build_url": {
+                                "type": "keyword"
+                            },
+                            "test_report_manifest_yml": {
+                                "type": "keyword"
+                            },
+                            "version": {
+                                "type": "keyword"
+                            },
+                            "component_test_spec_name": {
+                                "type": "keyword"
+                            },
+                            "component_test_spec_result": {
+                                "type": "keyword"
+                            },
+                            "component_test_spec_yml": {
+                                "type": "keyword"
+                            },
+                            "component_test_spec_test_stderr": {
+                                "type": "keyword"
+                            },
+                            "component_test_spec_test_stdout": {
+                                "type": "keyword"
+                            },
+                            "component_test_spec_cluster_stderr": {
+                                "type": "keyword"
+                            },
+                            "component_test_spec_cluster_stdout": {
+                                "type": "keyword"
+                            }
+                        }
+                    }
+                }'
+                curl -I "${METRICS_HOST_URL}/${indexName}" --aws-sigv4 \"aws:amz:us-east-1:es\" --user \"${awsAccessKey}:${awsSecretKey}\" -H \"x-amz-security-token:${awsSessionToken}\" | grep -E "HTTP\\/[0-9]+(\\.[0-9]+)? 200"
+                if [ \$? -eq 0 ]; then
+                    echo "Index already exists. Indexing Results"
+                else
+                    echo "Index does not exist. Creating..."
+                    create_index_response=\$(curl -s -XPUT "${METRICS_HOST_URL}/${indexName}" --aws-sigv4 \"aws:amz:us-east-1:es\" --user \"${awsAccessKey}:${awsSecretKey}\" -H \"x-amz-security-token:${awsSessionToken}\" -H 'Content-Type: application/json' -d "\${INDEX_MAPPING}")
```

Comment:
Maybe in future we can use a top level util package for this instead of curl commands with right initialization and checks? Wondering if it is possible to use java-client too.
@prudhvigodithi